### ISSUE TITLE
Permission does not include alias when attaching a function URL with a pre-traffic hook

### ISSUE THREAD

**Issue by johnc44** [aws/serverless-application-model#2386]
Adding FunctionUrlConfig to a serverless function definition which also has a pre-traffic hook creates a lambda permission that points to the latest version of the funciton, not the 'live' alias and browsing the lambda endpoint returns {"error\": "Forbidden"}.

Using the AWS Console, the lambda function shows the Permission on the latest version, but does not show the URL.  Changing to the live alias shows the URL but not the permission.

Example template:
`  MyTestLambda:
    Type: AWS::Serverless::Function
    Properties:
      CodeUri: src/
      Handler: main.lambda_handler
      Runtime: python3.9
      FunctionName: MyTestLambda
      Description: Test lambda
      AutoPublishAlias: live
      DeploymentPreference:
        Type: AllAtOnce
        Hooks:
          PreTraffic: !Ref PreTrafficHook

      FunctionUrlConfig:
        AuthType: NONE
`
For ref, this is what I had to add as a workaround:
`
    Type: AWS::Lambda::Permission
    Properties:
      Action: lambda:InvokeFunctionUrl
      Principal: "*"
      FunctionName:
        !Sub
          - "${name}:live"
          - name:
              !Ref myTestLambda
      FunctionUrlAuthType: NONE
`

Using aws-sam-cli version 1.46 on both macOS and Linux.

**Comment by moelasmar** [comment]
thanks @johnc44 for raising this issue, I confirmed that it is a bug. The created Permission should be like the following:
```
HelloWorldFunctionUrlPublicPermissions:
    Properties:
      Action: lambda:InvokeFunctionUrl
      FunctionName:
        Ref: HelloWorldFunctionAliaslive # refer to the alias, not the function
      FunctionUrlAuthType: NONE
      Principal: '*'
    Type: AWS::Lambda::Permission
```

instead of 
```
HelloWorldFunctionUrlPublicPermissions:
    Properties:
      Action: lambda:InvokeFunctionUrl
      FunctionName:
        Ref: HelloWorldFunction
      FunctionUrlAuthType: NONE
      Principal: '*'
    Type: AWS::Lambda::Permission
```

**Comment by moelasmar** [comment]
Pushed this [PR](https://github.com/aws/serverless-application-model/pull/2387) to fix this issue.

**Comment by johnc44** [comment]
Wow - that was fas!t. Thahks so much @[moelasmar](https://github.com/moelasmar)
One thing - when we added the permission we used a : between the function name and the alias (e.g. MyTestFunction:live).  I noticed you are appending "AliasLive" at the end (e.g. MyTestFunctionAliasLive).  I presume this has the same effect?

**Comment by hawflau** [comment]
Closing as the fix (#2378) has been release in v1.46.0
